Modules/_blake2/blake2module.h

```diff
@@ -0,0 +1,109 @@
+#ifndef Py_BLAKE2MODULE_H
+#define Py_BLAKE2MODULE_H
+
+#ifdef HAVE_LIBB2
+#include <blake2.h>
+
+// copied from blake2-impl.h
```

Comment:
carry the https://github.com/python/cpython/blob/main/Modules/_blake2/impl/blake2-impl.h license header text over as well.  Any reason we can't just `#include "impl/blaze2-impl.h"` instead? it seems to just contain these static inlines.

Reply:
Initially I removed the ``impl`` directory completely. Let's use ``impl/blaze2-impl.h`` until we stop vendoring blake2 completely.